test/e2e/packetcapture_test.go

```diff
@@ -670,3 +706,65 @@ func conditionSliceEqualsIgnoreLastTransitionTime(as, bs []crdv1alpha1.PacketCap
 	}
 	return true
 }
+
+// verifyPacketFile will read the packets file and check if packet count and packet data match with CR.
+func verifyPacketFile(t *testing.T, pc *crdv1alpha1.PacketCapture, reader io.Reader, targetNum int32, srcIP net.IP, dstIP net.IP) (err error) {
+	ngReader, err := pcapgo.NewNgReader(reader, pcapgo.DefaultNgReaderOptions)
+	if err != nil {
+		return err
+	}
+
+	for i := int32(0); i < targetNum; i++ {
+		data, _, err := ngReader.ReadPacketData()
+		if err != nil {
+			return err
+		}
+		packet := gopacket.NewPacket(data, layers.LayerTypeEthernet, gopacket.Default)
+		ipLayer := packet.Layer(layers.LayerTypeIPv4)
+		require.NotNil(t, ipLayer)
+		ip, _ := ipLayer.(*layers.IPv4)
+		assert.Equal(t, srcIP.String(), ip.SrcIP.String())
+		assert.Equal(t, dstIP.String(), ip.DstIP.String())
+
+		if pc.Spec.Packet == nil {
+			return nil
+		}
+
+		packetSpec := pc.Spec.Packet
+		proto := packetSpec.Protocol
+		if proto == nil {
+			return nil
```

Comment:
I think the intention of the test is to validate all packets, not just the 1st one? Shouldn't it be `continue`?

Reply:
yes. updated.